homeassistant/components/media_player/squeezebox.py

```diff
@@ -60,129 +66,111 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
                       host, port, error)
         return False
 
-    # Combine it with port to allow multiple servers at the same host
-    key = "{}:{}".format(ipaddr, port)
```

Comment:
Ah, I see. It sounds like the auto-discovery is the real root problem then. Can netdisco be updated to supply the port as well? It looks like it's only returning the hostname here: https://github.com/home-assistant/netdisco/blob/master/netdisco/discoverables/sonos.py#L12 If this is updated to return a tuple hostname/port like the chromecast that should solve the issue. I don't have a sonos setup to test.